java/vector/src/main/java/org/apache/arrow/vector/VectorLoader.java

```diff
@@ -79,7 +105,12 @@ private void loadBuffers(
     List<ArrowBuf> ownBuffers = new ArrayList<>(bufferLayoutCount);
     for (int j = 0; j < bufferLayoutCount; j++) {
       ArrowBuf nextBuf = buffers.next();
-      ownBuffers.add(codec.decompress(vector.getAllocator(), nextBuf));
+      // for vectors without nulls, the buffer is empty, so there is no need to decompress it.
+      ArrowBuf bufferToAdd = nextBuf.writerIndex() > 0 ? codec.decompress(vector.getAllocator(), nextBuf) : nextBuf;
+      ownBuffers.add(bufferToAdd);
+      if (decompressionPerformed) {
+        decompressedBuffers.add(bufferToAdd);
+      }
     }
     try {
       vector.loadFieldBuffers(fieldNode, ownBuffers);
```

Comment:
could we close decompressed buffers after loading them here?

Reply:
Good suggestion. This way, we do not need to change the public interface of `VectorLoader`.